src/classes/frDonation.cls

```diff
@@ -77,10 +77,12 @@ public class frDonation extends frModel {
         this.setDonorId(String.valueOf(request.get('donorId')));
         String status = String.valueOf(request.get('status'));
         
-        if (Boolean.valueOf(request.get('pledge'))) {
+        if(String.isBlank(o.StageName)) {
             if (status == 'Pending') {
                 o.StageName = 'Pending';
-            } else if (status == 'Closed') {
+            } else if (status == 'Failed' || status == 'Refunded') {
```

Comment:
Is it possible to have a single place of constants where we can pull these strings from?

Reply:
fixed this as well